components/camel-spring-ws/src/main/java/org/apache/camel/component/spring/ws/SpringWebserviceProducer.java

```diff
@@ -139,7 +149,18 @@ private void populateMessageHeaderFromResponse(Message message, SoapHeader soapH
 
         }
     }
-
+    /**
+     * Populates message attachments from soap response attachments 
+     * @param inOrOut {@link Message}
+     * @param soapMessage {@link SoapMessage}
+     */
+    private void populateMessageAttachmentsFromResponse(Message inOrOut, Iterator<Attachment> attachments) {
```

Comment:
I made those fixes you requested, even though i did not implemented the soap header part, i just added the attachment population. ;)

But anyway, let me know if you have any other change requests.

Regards Marcus